test/mysql2_test.rb

```diff
@@ -160,6 +160,26 @@ def test_resource_acquisition_for_query
     end
   end
 
+  def test_semian_allows_rollback
+    client = connect_to_mysql!
+
+    client.query('START TRANSACTION;')
+
+    Semian[:mysql_testing].acquire do
+      client.query('ROLLBACK;')
+    end
+  end
```

Comment:
@Sirupsen `Mysql::Client#query` is monkey patched to call acquire too.

So this come down to:

```ruby
acquire do
  acquire do
    original_query
  end
end
```